pkg/tar/tar.go

```diff
@@ -198,6 +198,9 @@ func (t *stiTar) CreateTarStreamWithLogging(dir string, includeDirInPath bool, w
 		if !info.IsDir() && !t.shouldExclude(path) {
 			// if file is a link just writing header info is enough
 			if info.Mode()&os.ModeSymlink != 0 {
+				if dir == path {
```

Comment:
is this only a problem when the root dir is a symlink?  or do we need to also address symlinks within the root dir that are pointing to dirs?